src/app/logic/petition/sign-petition.service.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable } from '@angular/core';
+import {
+  exhaustMap,
+  map,
+  merge,
+  Observable,
+  partition,
+  shareReplay,
+  Subject,
+  tap,
+} from 'rxjs';
+import { Result } from 'src/app/shared/models/exports';
+import { SignaturePetitionData } from 'src/app/shared/models/petition/signature-petition-data';
+import { PetitionService } from './petition.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SignPetitionService {
```

Comment:
This service seems to be a state management one. Please don't provide it in root.